### PR TITLE
Player API

### PR DESCRIPTION
**Pull request by R-Wang97**
Changelog:

1. Completed functions for player api
2. Modified player model to have function for parsing array to json
3. TODO: Add endpoints for player api functions (not sure what to call endpoints)

### CODE REVIEW

File: store/player_store.go [hunk 1]
```diff
@@ -167,6 +167,26 @@ func (ps PlayerStore) GetAll() StoreChannel {
...
 }
 
+func (ps PlayerStore) GetPlayerGames(playerId string) StoreChannel {
+	storeChannel := make(StoreChannel)
+
+	go func() {
+		result := StoreResult{}
+
+		var data []*model.Game
+		if _, err := ps.GetMaster().Select(&data, "SELECT * FROM Players WHERE PlayerId = :PlayerId", map[string]interface{}{"PlayerId": playerId}); err != nil {
```
Comment (DavidLu1997) [inline store/player_store.go]: 90% sure its stored as just `id`
Check stores/player_store.go

File: api/player.go [hunk 2]
```diff
@@ -62,15 +64,79 @@ func updatePlayer(s *Session, w http.ResponseWriter, r *http.Request) {
...
+	email := params["username"]	
+
+	if result, err := GetPlayer(username); err != nil {
+		s.Err = err
+	} else {
+		w.Write([]byte(result.ToJson(result)))
+	}	
+}
+
+func GetUsername(username string) (*model.Player, *model.Error) {
```
Comment (DavidLu1997) [inline api/player.go]: Should rename to `GetByUsername`

File: api/player.go [hunk 3]
```diff
@@ -62,15 +64,79 @@ func updatePlayer(s *Session, w http.ResponseWriter, r *http.Request) {
...
+	email := params["email"]	
 
+	if result, err := GetPlayer(email); err != nil {
+		s.Err = err
+	} else {
+		w.Write([]byte(result.ToJson(result)))
+	}	
+}
+
+func GetEmail(email string) (*model.Player, *model.Error) {
```
Comment (DavidLu1997) [inline api/player.go]: Should rename to `GetByEmail`

File: api/player.go [hunk 4]
```diff
@@ -17,6 +18,7 @@ func InitPlayer() {
 	BaseRoutes.NeedPlayer.Handle("/get", ApiPlayerRequired(getPlayer)).Methods("GET")
 	BaseRoutes.NeedPlayer.Handle("/games", ApiPlayerRequired(getPlayerGames)).Methods("GET")
 	BaseRoutes.NeedPlayer.Handle("/find", ApiPlayerRequired(findPlayer)).Methods("POST")
+	//Need to add endpoints for getByEmail and getByUsername
```
Comment (DavidLu1997) [inline api/player.go]: `BaseRoutes.Player.Handle("/get_email/{email:[A-Za-z0-9+.@]}", ApiPlayer(getPlayerByEmail)).Methods("GET")`

File: api/player.go [hunk 5]
```diff
@@ -8,15 +8,16 @@ import (
...
 )
 
 func InitPlayer() {
 	BaseRoutes.Players.Handle("/create", ApiHandler(createPlayer)).Methods("POST")
-
 	BaseRoutes.NeedPlayer.Handle("/update", ApiPlayerRequired(updatePlayer)).Methods("POST")
 	BaseRoutes.NeedPlayer.Handle("/get", ApiPlayerRequired(getPlayer)).Methods("GET")
 	BaseRoutes.NeedPlayer.Handle("/games", ApiPlayerRequired(getPlayerGames)).Methods("GET")
-	BaseRoutes.NeedPlayer.Handle("/find", ApiPlayerRequired(findPlayer)).Methods("POST")
+	BaseRoutes.Player.Handle("/get_email/{email:[A-Za-z0-9+.@]}", ApiPlayer(getPlayerByEmail)).Methods("GET")
```
Comment (DavidLu1997) [inline api/player.go]: lets require a player for this

**Comment by DavidLu1997** [comment]
+1 once it builds
@Denton-L review

File: api/player.go [hunk 6]
```diff
@@ -8,15 +8,16 @@ import (
...
 )
 
 func InitPlayer() {
 	BaseRoutes.Players.Handle("/create", ApiHandler(createPlayer)).Methods("POST")
-
 	BaseRoutes.NeedPlayer.Handle("/update", ApiPlayerRequired(updatePlayer)).Methods("POST")
 	BaseRoutes.NeedPlayer.Handle("/get", ApiPlayerRequired(getPlayer)).Methods("GET")
 	BaseRoutes.NeedPlayer.Handle("/games", ApiPlayerRequired(getPlayerGames)).Methods("GET")
-	BaseRoutes.NeedPlayer.Handle("/find", ApiPlayerRequired(findPlayer)).Methods("POST")
+	BaseRoutes.NeedPlayer.Handle("/get_email/{email:[A-Za-z0-9+.@]}", ApiPlayerRequired(getPlayerByEmail)).Methods("GET")
```
Comment (ian952) [inline api/player.go]: Characters such as _ or - is not considered valid characters

Comment (DavidLu1997) [inline api/player.go]: @ian952 want to fix it?

Comment (Denton-L) [inline api/player.go]: No, I think `_` and `-` are valid characters.

Source: https://stackoverflow.com/questions/1547899/which-characters-make-a-url-invalid

Comment (ian952) [inline api/player.go]: Regex currently marks `_` and `-` as invalid

Comment (Denton-L) [inline api/player.go]: They're both valid but I think using `-` is more standard practice.

Comment (DavidLu1997) [inline api/player.go]: Let's open an issue for now

Comment (DavidLu1997) [inline api/player.go]: Anyone want to rewrite this, ish? @Denton-L

**Pull request merged**